nautobot/utilities/templatetags/helpers.py

```diff
@@ -123,7 +123,7 @@ def render_yaml(value):
     """
     Render a dictionary as formatted YAML.
     """
-    return yaml.dump(json.loads(json.dumps(value)))
+    return yaml.dump(json.loads(json.dumps(value, ensure_ascii=False)), allow_unicode=True)
```

Comment:
Also it's curious to me that this function does *not* set a value for `sort_keys` where `render_json` does - seems inconsistent to me.

Reply:
It was done to address ordered dictionaries dumping extra unexpected data in the output: https://github.com/netbox-community/netbox/issues/4150

Reply:
```py
>>> od = OrderedDict()
>>> od['a'] = 1
>>> od['b'] = 2
>>> od['c'] = 3
>>> print(yaml.dump(od))
!!python/object/apply:collections.OrderedDict
- - - a
    - 1
  - - b
    - 2
  - - c
    - 3
```

Reply:
Gross. Thanks for the context, and more evidence that we should review places where we're using OrderedDict and see where they're no longer necessary now that vanilla dicts preserve insertion order.

Reply:
What's odd is the previous call before that change should have worked: https://github.com/netbox-community/netbox/blob/2ab382eec5a6a282bc99209073690d9c19cb55a8/netbox/utilities/templatetags/helpers.py#L81-L85

```py
>>> od = OrderedDict()
>>> od['a'] = 1
>>> od['b'] = 2
>>> od['c'] = 3
>>> print(yaml.dump(od))
!!python/object/apply:collections.OrderedDict
- - - a
    - 1
  - - b
    - 2
  - - c
    - 3

>>> print(yaml.dump(dict(od)))
a: 1
b: 2
c: 3

>>> d = {}
>>> d['a'] = 1
>>> d['b'] = 2
>>> d['c'] = 3
>>> print(yaml.dump(dict(d)))
a: 1
b: 2
c: 3
```

Reply:
Possibly not in the case of nested OrderedDicts?

Reply:
> Gross. Thanks for the context, and more evidence that we should review places where we're using OrderedDict and see where they're no longer necessary now that vanilla dicts preserve insertion order.

Yes since we've dropped 3.6 now and 3.7+ has insertion order in dictionaries as a guaranteed language feature we should be fine to drop `OrderedDict`.

Question: seeing my comment above, should we revert to using `yaml.dump(dict(value))`?

Reply:
Python 3.6 implemented the order-preserving of vanilla dicts: https://docs.python.org/3/whatsnew/3.6.html#new-dict-implementation

So we don't actually need to use OrderedDict as we don't support 3.7 or below. Should we create a ticket to remove all uses of OrderedDict? Or do you think we use other functionality of OrderedDict other than its inherent ordering capability? It seems there are memory wins by just using the vanilla dict.

https://realpython.com/python-ordereddict/#selecting-the-right-dictionary-for-the-job

Reply:
OK cool I'll open the issue

Reply:
Ah sorry @bryanculver I didn't see your comment before! Oops!

Reply:
> Also it's curious to me that this function does _not_ set a value for `sort_keys` where `render_json` does - seems inconsistent to me.

This doesn't seem to be necessary, as it appear they are being sorted in `yaml.dump()`:

```py
>>> d = {"a": 1, "d": 2, "c": 5}

>>> json.dumps(d, indent=4, sort_keys=True)
'{\n    "a": 1,\n    "c": 5,\n    "d": 2\n}'

>>> print(json.dumps(d, indent=4, sort_keys=True))
{
    "a": 1,
    "c": 5,
    "d": 2
}
>>> print(json.dumps(d, indent=4))
{
    "a": 1,
    "d": 2,
    "c": 5
}
>>> print(json.loads(json.dumps(d, indent=4)))
{'a': 1, 'd': 2, 'c': 5}
>>> print(yaml.dump(json.loads(json.dumps(d, indent=4))))
a: 1
c: 5
d: 2
```

Reply:
One further possibility to try might be `yaml.safe_dump`?